homeassistant/components/osoenergy/__init__.py

```diff
@@ -0,0 +1,140 @@
+"""Support for the OSO Energy devices and setrvices."""
+from functools import wraps
+import logging
+
+from aiohttp.web_exceptions import HTTPException
+from apyosoenergyapi import OSOEnergy
+from apyosoenergyapi.helper.osoenergy_exceptions import OSOEnergyReauthRequired
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN, PLATFORM_LOOKUP, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY): cv.string,
+                    vol.Optional(CONF_SCAN_INTERVAL, default=15): cv.positive_int,
+                },
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
```

Comment:
This integration communicates with a device/service and should not have YAML configuration.
<https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md>

Reply:
Removed the CONFIG_SCHEMA